visualization/app/codeCharta/util/treeMapGenerator.ts

```diff
@@ -25,56 +38,80 @@ export function createTreemapNodes(map: CodeMapNode, state: State, metricData: N
 }
 
 function buildSquarifiedTreeMapsForFixedFolders(
-	map: CodeMapNode,
+	hierarchyNode: HierarchyNode<CodeMapNode>,
 	state: State,
+	scaleLength: number,
+	scaleWidth: number,
+	offsetX0: number,
+	offsetY0: number,
 	heightScale: number,
 	maxHeight: number,
 	isDeltaState: boolean
 ) {
-	const hierarchyNode = hierarchy(map)
+	const nodes = []
 
-	const nodes: Node[] = [TreeMapHelper.buildRootFolderForFixedFolders(map, heightScale, state, isDeltaState)]
-	const scale = (state.treeMap.mapSize * 2 + getEstimatedNodesPerSide(hierarchyNode) * state.dynamicSettings.margin) / nodes[0].length
+	for (const fixedFolder of hierarchyNode.children) {
 
-	scaleRoot(nodes[0], scale)
+		const squarified = getSquarifiedTreeMap(fixedFolder.data, state)
 
-	for (const fixedFolder of map.children) {
-		const squarified = getSquarifiedTreeMap(fixedFolder, state)
 		for (const squarifiedNode of squarified.treeMap.descendants()) {
-			scaleAndTranslateSquarifiedNode(squarifiedNode, fixedFolder, squarified, scale)
+			// squarified.width/height is a sum of the fixedPosition.width/height and applied margins
+			// The following scaling factors are used later to calculate out the original percentage fixed width/height.
+			// Example: Width and Height fixed at 20. Added margins 42px (absolute value) for each = 62px
+			//  20/62 = 0.322...       62 x 0.322... = 20 :)
+			const scaleX = fixedFolder.data.fixedPosition.width / squarified.width
+			const scaleY = fixedFolder.data.fixedPosition.height / squarified.height
```

Comment:
I would store fixedFolders.data.fixedPosition in a variable as we use it quite often, here and in lines 66-70.